src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useState, useEffect, useContext } from 'react';
+import type { ReactNode } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+interface AuthContextType {
+  isLoggedIn: boolean;
+  login: (token: string) => void;
+  logout: () => void;
+}
+
+const AuthContext = createContext<AuthContextType | null>(null);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(() => !!sessionStorage.getItem('authToken'));
```

Comment:
useState<boolean> 을 쓰지 않아도 typescript가 자연스럽게 추론할 수 있어용!
```suggestion
  const [isLoggedIn, setIsLoggedIn] = useState(() => !!sessionStorage.getItem('authToken'));
```